### PR TITLE
[adsp] show warning if no modes are present for manager dialog

### PR DESCRIPTION
**Pull request by AlwinEsch**
This fix a bad dialog if no modes are present for DSP manager. It becomes checked for available modes, if nothing present becomes a warning shown and the dialog closed again.

screenshot8

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -8026,7 +8026,11 @@ msgctxt "#15064"
 msgid "Apply changes directly"
 msgstr ""
 
-#empty string with id 15065
+#. Warning if no dsp modes are present
+#: xbmc/settings/dialogs/GUIDialogAudioDSPManager.cpp
+msgctxt "#15065"
+msgid "No audio DSP modes present!"
```
Comment (phil65) [inline addons/resource.language.en_gb/resources/strings.po]: What do you mean exactly here?  "No audio DSP modes active"? "No audio DSP modes available"?

Comment (AlwinEsch) [inline addons/resource.language.en_gb/resources/strings.po]: It means "No audio DSP modes available", to inform user during open of manager dialog and close them directly after OK Dialog with this message.

Better change it to your given text?

Comment (phil65) [inline addons/resource.language.en_gb/resources/strings.po]: Yes, I think "available" is better. Wouldnt it be possible to not open the manager dialog at all then?

Comment (AlwinEsch) [inline addons/resource.language.en_gb/resources/strings.po]: Yes I can prevent the open with the use on:

```cpp
void CActiveAEDSP::OnSettingAction(const CSetting *setting)
{
  if (setting == NULL)
    return;

  const std::string &settingId = setting->GetId();

  if (settingId == CSettings::SETTING_AUDIOOUTPUT_DSPSETTINGS)
  {
    if (IsActivated())
    {
      CGUIDialogAudioDSPManager *dialog = (CGUIDialogAudioDSPManager *)g_windowManager.GetWindow(WINDOW_DIALOG_AUDIO_DSP_MANAGER);
      if (dialog)
        dialog->Open();
    }
  }
  else if (settingId == CSettings::SETTING_AUDIOOUTPUT_DSPRESETDB)
  {
    if (HELPERS::ShowYesNoDialogLines(CVariant{19098}, CVariant{36440}, CVariant{750}) ==
      DialogResponse::YES)
    {
      CDateTime::ResetTimezoneBias();
      ResetDatabase();
    }
  }
}
```

And already planing to hide the button on Estuary if no modes are available.

**Comment by AlwinEsch** [comment]
Have changed it, now the dialog becomes show before the manager becomes opened.
Also becomes the manager button on OSD hidden without modes present. For them I has added the gui info manager option with `ADSP.HasModes`.

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
OK to merge the fix?

**Comment by FernetMenta** [comment]
who do you ask? I can't give much input here.

**Comment by AlwinEsch** [comment]
More or less to all to have minimum one +1 or a note for improvement.

**Comment by FernetMenta** [comment]
from what I can tell it looks ok, +1

**Pull request merged**